hopp/simulation/technologies/generic/generic_plant.py

```diff
@@ -0,0 +1,351 @@
+
+    """
+
+    system_capacity_kw: float = field(default = 0.0)
+    system_capacity_kwac: Optional[float] = field(default = 0.0)
+    generation_profile_kw: Optional[list[float]] = field(default = None)
+    subsystem_name: Optional[str] = field(default="generic_system")
+
+    n_timesteps: Union[float,int] = field(default = 8760)
+    fin_model: Optional[Union[dict, FinancialModelType]] = field(default=None)
+    name: str = field(default="GenericPlant")
+    
+    
+
+@define 
+class GenericSystem(BaseClass):
+    """Represents a single generic system defined by its system capacity and generation profile.
+
+    Args:
+        system_capacity (float): system capacity in kW. For DC-systems, 
+            this is likely equal to the system capacity in kW-DC.
+        gen (list[float]): generation profile in kW. 
+        system_capacity_ac (float, Optional): system capacity in kW-AC. Defaults to system_capacity if not specified.
+            Should be specified if `system_capacity_ac` is different than `system_capacity`. 
+            This input is helpful when representing DC-systems such as ``PVPlant``.
+        system_name (str, Optional): name of system, primarily used if using `GenericMultiSystem`.
+            Defaults to "generic_system".
+        n_timesteps (float | int, Optional): number of timesteps in a simulation. Defaults to 8760.
+        t_step (float | int, Optional): time step in hours. Defaults to 1.
+
+    Attributes:
+        annual_energy (float): annual energy production in kWh/year
+        capacity_factor (float): capacity factor of the system based on system_capacity as a percent
+        annual_energy_pre_curtailment_ac (float): annual energy production in kWh/year
+    """
+
+    system_capacity: float = field(validator=gt_zero)
+    gen: list[float]
+    
+    system_capacity_ac: Optional[float] = field(default = 0.0)
+    system_name: Optional[str] = field(default = "generic_system")
+    n_timesteps: Optional[Union[float,int]] = field(default = 8760)
+    t_step: Optional[Union[float,int]] = field(default = 1)
+    
+    # Calculated values
+    annual_energy: float = field(init = False)
+    capacity_factor: float = field(init = False)
+    annual_energy_pre_curtailment_ac: float = field(init = False)
+    
+    def __attrs_post_init__(self):
+        """Initialize some attributes and set defaults if needed. This method does the following:
+
+        1) calculate attributes:
+
+            - `annual_energy`
+
+            - `annual_energy_pre_curtailment_ac`
+
+            - `capacity_factor`
+
+        2) set `system_capacity_ac` to `system_capacity` if `system_capacity_ac` was not input.
+
+        Raises:
+            ValueError: if length of self.gen is not equal to self.n_timesteps
+        """
+
+        self.annual_energy = np.sum(self.gen)
+        self.annual_energy_pre_curtailment_ac = np.sum(self.gen)
+        
+        if self.system_capacity_ac==0.0:
```

Comment:
Non-blocking nitpick: could you make the default `None` and then have this check be `if self.system_capacity_ac is None` so it's very clear that this is just overriding the default?